src/solvers/dgsem/basis_lobatto_legendre.jl

```diff
@@ -196,47 +214,79 @@ function Base.show(io::IO, ::MIME"text/plain", mortar::LobattoLegendreMortarL2)
+                                                   reverse_upper, reverse_lower)
+end
+
+function Base.show(io::IO, mortar::LobattoLegendreMortarEC)
+    @nospecialize mortar # reduce precompilation time
+
+    print(io, "LobattoLegendreMortarEC{", real(mortar), "}(polydeg=", polydeg(mortar),
+          ")")
+end
+function Base.show(io::IO, ::MIME"text/plain", mortar::LobattoLegendreMortarEC)
+    @nospecialize mortar # reduce precompilation time
+
+    print(io, "LobattoLegendreMortarEC{", real(mortar), "} with polynomials of degree ",
+          polydeg(mortar))
+end
+
+@inline Base.real(mortar::Union{LobattoLegendreMortarL2{RealT},
+LobattoLegendreMortarEC{RealT}}) where {RealT} = RealT
 
-@inline function nnodes(mortar::LobattoLegendreMortarL2{RealT, NNODES}) where {RealT,
-                                                                               NNODES}
+@inline function nnodes(mortar::Union{LobattoLegendreMortarL2{RealT, NNODES},
+                                      LobattoLegendreMortarEC{RealT, NNODES}}) where {
+                                                                                      RealT,
+                                                                                      NNODES
+                                                                                      }
     NNODES
 end
 
-@inline polydeg(mortar::LobattoLegendreMortarL2) = nnodes(mortar) - 1
-
-# TODO: We can create EC mortars along the lines of the following implementation.
-# abstract type AbstractMortarEC{RealT} <: AbstractMortar{RealT} end
-
-# struct LobattoLegendreMortarEC{RealT<:Real, NNODES, MortarMatrix<:AbstractMatrix{RealT}, SurfaceFlux} <: AbstractMortarEC{RealT}
-#   forward_upper::MortarMatrix
-#   forward_lower::MortarMatrix
-#   reverse_upper::MortarMatrix
-#   reverse_lower::MortarMatrix
-#   surface_flux::SurfaceFlux
-# end
-
-# function MortarEC(basis::LobattoLegendreBasis{RealT}, surface_flux)
-#   forward_upper   = calc_forward_upper(n_nodes, RealT)
-#   forward_lower   = calc_forward_lower(n_nodes, RealT)
-#   l2reverse_upper = calc_reverse_upper(n_nodes, Val(:gauss_lobatto), RealT)
-#   l2reverse_lower = calc_reverse_lower(n_nodes, Val(:gauss_lobatto), RealT)
-
-#   # type conversions to make use of StaticArrays etc.
-#   nnodes_ = nnodes(basis)
-#   forward_upper   = SMatrix{nnodes_, nnodes_}(forward_upper)
-#   forward_lower   = SMatrix{nnodes_, nnodes_}(forward_lower)
-#   l2reverse_upper = SMatrix{nnodes_, nnodes_}(l2reverse_upper)
-#   l2reverse_lower = SMatrix{nnodes_, nnodes_}(l2reverse_lower)
-
-#   LobattoLegendreMortarEC{RealT, nnodes_, typeof(forward_upper), typeof(surface_flux)}(
-#     forward_upper, forward_lower,
-#     l2reverse_upper, l2reverse_lower,
-#     surface_flux
-#   )
-# end
-
-# @inline nnodes(mortar::LobattoLegendreMortarEC{RealT, NNODES}) = NNODES
+@inline polydeg(mortar::Union{LobattoLegendreMortarL2,
+LobattoLegendreMortarEC}) = nnodes(mortar) - 1
```

Comment:
This is quite ugly formatting :(

Reply:
```suggestion
@inline function polydeg(mortar::Union{LobattoLegendreMortarL2, LobattoLegendreMortarEC})
    return nnodes(mortar) - 1
end
```

But see also above (define union constant)